### PR TITLE
MNIOR: feat(apps): support config file

### PR DESCRIPTION
**Pull request by sushi30**
- added support for app config files
- removed AppPrivateConfig from the OpenMetadata server configuration

### Describe your changes:

Fixes <issue-number>

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/apps/ConfigurationReader.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
+package org.openmetadata.service.apps;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import joptsimple.internal.Strings;
```
Comment (mohityadav766) [inline openmetadata-service/src/main/java/org/openmetadata/service/apps/ConfigurationReader.java:9]: @sushi30 not sure if it works , but we can do something like below a try

```
     ObjectMapper objectMapper = Jackson.newObjectMapper();
    Validator validator = Validators.newValidator();
    YamlConfigurationFactory<AppPrivateConfig> factory =
        new YamlConfigurationFactory<>(
                AppPrivateConfig.class, validator, objectMapper, "dw");
    config =
        factory.build(
            new SubstitutingSourceProvider(
                new FileConfigurationSourceProvider(), new EnvironmentVariableSubstitutor(false)),
            configFilePath);
```
We are using this in Dropwizard

Comment (sushi30) [reply openmetadata-service/src/main/java/org/openmetadata/service/apps/ConfigurationReader.java]: addressed

File: openmetadata-service/src/main/java/org/openmetadata/service/apps/ConfigurationReader.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import java.util.Map;
+import org.apache.commons.text.StringSubstitutor;
+import org.openmetadata.schema.api.configuration.apps.AppPrivateConfig;
+import org.openmetadata.service.util.JsonUtils;
+
+public class ConfigurationReader {
+
+  // envMap is for custom environment variables (e.g., for testing), defaulting to the system
+  // environment.
+  private final Map<String, String> envMap;
```
Comment (mohityadav766) [inline openmetadata-service/src/main/java/org/openmetadata/service/apps/ConfigurationReader.java:22]: ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
    YamlConfigurationFactory<Object> factory =
        new YamlConfigurationFactory<>(Object.class, null, mapper, "dw");
    StringSubstitutor substitutor =
        this.envMap == null
            ? new EnvironmentVariableSubstitutor(false)
            : new StringSubstitutor(this.envMap);
            
            these can be initialize once

Comment (sushi30) [reply openmetadata-service/src/main/java/org/openmetadata/service/apps/ConfigurationReader.java]: done

**Review by mohityadav766 — APPROVED** [review]

**Pull request merged**